test/integration/create-next-app/templates/matrix.test.ts

```diff
@@ -33,24 +33,30 @@ describe.each(['app', 'pages'] as const)(
       // eslint: ['--eslint', '--no-linter'],
       eslint: ['--eslint'],
 
-      srcDir: ['--src-dir', '--no-src-dir'],
+      // Trading test perf for robustness:
+      // srcDir and reactCompiler don't interact so we're testing them together
+      // instead of all permutations.
+      srcDirAndCompiler: [
+        '--src-dir --react-compiler',
+        '--no-src-dir --no-react-compiler',
+      ],
```

Comment:
Going from 24 to 48 test cases seemed excessive to me. If we regress on a specific combination we can add it manually.